code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -2383,3 +2444,60 @@
 		M.adjustOxyLoss(2, FALSE)
 		M.adjustBruteLoss(2, FALSE)
 	..()
+
+/datum/reagent/acidic_buffer
+	name = "Strong acidic buffer"
+	description = "This reagent will consume itself and move the pH of a beaker towards acidity when added to another."
+	color = "#fbc314"
+	pH = 0
+	data = list("pH" = 0)
+	var/strength = 1
+
+//Consumes self on addition and shifts pH
+/datum/reagent/acidic_buffer/on_new(list/data)
+	. = ..()
+	if(!holder)
+		return 
+	if(holder.flags & NO_REACT)
+		return
+	if(holder.has_reagent(/datum/reagent/stabilizing_agent))
+		return 
+	if(LAZYLEN(holder.reagent_list) == 1)
+		return
+	if(holder.pH < pH)
+		holder.my_atom.visible_message("<span class='warning'>The beaker fizzes as the buffer is added, to no effect.</b></span>")
+		playsound(holder.my_atom, 'sound/chemistry/bufferadd.ogg', 50, TRUE)
+		return
+	holder.pH = clamp((((holder.pH * (holder.total_volume-(volume*strength)))+(pH * (volume*strength)) )/holder.total_volume), 0, 14) //This is BEFORE removal
+	holder.my_atom.visible_message("<span class='warning'>The beaker fizzes as the pH changes!</b></span>")
+	playsound(holder.my_atom, 'sound/chemistry/bufferadd.ogg', 50, TRUE)
+	holder.remove_reagent(type, volume, ignore_pH = TRUE)
+	
+
+/datum/reagent/basic_buffer
+	name = "Strong basic buffer"
+	description = "This reagent will consume itself and move the pH of a beaker towards alkalinity when added to another."
+	color = "#3853a4"
+	pH = 14
+	data = list("pH" = 14)
+	var/strength = 1
+
+/datum/reagent/basic_buffer/on_new(list/data)
+	. = ..()
+	if(!holder)
+		return 
+	if(holder.flags & NO_REACT)
+		return
+	if(holder.has_reagent(/datum/reagent/stabilizing_agent))
+		return 
+	if(LAZYLEN(holder.reagent_list) == 1)
+		return
+	if(holder.pH > pH)
+		holder.my_atom.visible_message("<span class='warning'>The beaker froths as the buffer is added, to no effect.</b></span>")
+		playsound(holder.my_atom, 'sound/chemistry/bufferadd.ogg', 50, TRUE)
+		return 
+	holder.pH = clamp((((holder.pH * (holder.total_volume-(volume*strength)))+(pH * (volume*strength)) )/holder.total_volume), 0, 14)
```

Comment:
`holder.total_volume` can never be zero at this point, right?

Reply:
This should be called during the add_reagent() method, which scrubs any negative or 0 additions. As this is added it checks to see if it's not the only one in the list, and if it is it returns ..(). I would be very surprised if holder.total_volume ever was 0 with 2 or more reagent datums in it, but I suppose it's not impossible, but I would expect it to be the result of a larger problem. I'm happy to add a check if you recommend it.

Reply:
I reworked how pH was done from feedback, this isn't a concern anymore.